core/src/test/java/apoc/util/UtilQuoteTest.java

```diff
@@ -0,0 +1,62 @@
+package apoc.util;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.neo4j.test.rule.DbmsRule;
+import org.neo4j.test.rule.ImpermanentDbmsRule;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assume.assumeTrue;
+
+@RunWith(Parameterized.class)
+public class UtilQuoteTest {
+
+    @ClassRule
+    public static DbmsRule db = new ImpermanentDbmsRule();
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> parameters() {
+        return Arrays.asList(new Object[][] {
+                { "abc", true },
+                { "_id", true },
+                { "some_var", true },
+                { "$lock", false },
+                { "has$inside", true },
+                { "ähhh", true },
+                { "rübe", true },
+                { "rådhuset", true },
+                { "1first", false },
+                { "first1", true },
+                { "a weird identifier", false },
+                { "^n", false },
+                { "$$n", false },
+                { " $var ", false },
```

Comment:
Thanks for pointing me towards this.

My intention was to have a test case for an identifier with a leading space because that's a valid identifier when quoted. But I just realized I should have rather used ` var ` as a test case for that. And then the test method needs fixing... That's what I just pushed.

Or did you mean something else?